### ISSUE TITLE
Request for fix: false positive in MissingFieldInEquals

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#74]
False positive in following code:

```
class ErrorInfo {

    private final ErrorCode errorCode;

    public ErrorInfo(final String errorCode) {
        this.errorCode = ErrorCode.valueOf(errorCode);
    }
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        ErrorInfo errorInfo = (ErrorInfo) o;
        return errorCode == errorInfo.errorCode;
    }
}
```
errorCode is the only field and ErrorCode is an enum.

**Comment by jborgers** [comment]
Now checks for fields assigned to in a simple way, if they are present in the equals
